mycelo/config/config.go

```diff
@@ -0,0 +1,191 @@
+package config
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math/big"
+	"os"
+
+	"github.com/ethereum/go-ethereum/internal/fileutils"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type Environment struct {
+	Paths         Paths
+	EnvConfig     EnvConfig
+	GenesisConfig GenesisConfig
+
+	// Derived Fields
+	accounts *GenesisAccounts
+}
+
+func NewEnv(envpath string, cfg *EnvConfig) (*Environment, error) {
+	env := &Environment{
+		Paths:         Paths{Workdir: envpath},
+		EnvConfig:     *cfg,
+		GenesisConfig: *BaseContractsConfig(),
+	}
+
+	accounts, err := createGenesisAccounts(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	env.accounts = accounts
+	return env, nil
+}
+
+func (env *Environment) WriteGenesisConfig() error {
+	if !fileutils.FileExists(env.Paths.Workdir) {
+		os.MkdirAll(env.Paths.Workdir, os.ModePerm)
+	}
+
+	if err := writeJson(env.GenesisConfig, env.Paths.GenesisConfig()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (env *Environment) WriteEnvConfig() error {
+	if !fileutils.FileExists(env.Paths.Workdir) {
+		os.MkdirAll(env.Paths.Workdir, os.ModePerm)
+	}
+
+	if err := writeJson(env.EnvConfig, env.Paths.EnvConfig()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ReadBuildEnv(envpath string) (*Environment, error) {
+	env := &Environment{
+		Paths: Paths{Workdir: envpath},
+	}
+
+	if err := readJson(&env.EnvConfig, env.Paths.EnvConfig()); err != nil {
+		return nil, err
+	}
+
+	if err := readJson(&env.GenesisConfig, env.Paths.GenesisConfig()); err != nil {
```

Comment:
in our new schema this can definitely fail. You don't need the genesis-config.json on the env. In fact, if you create the env directly from template we may not even generate this file

Reply:
See `ReadEnv` (it is true that pre-genesis and post-genesis configs stuff is mixed together)